server/channels/app/reaction.go

```diff
@@ -20,6 +20,15 @@ func (a *App) SaveReactionForPost(c request.CTX, reaction *model.Reaction) (*mod
 		return nil, err
 	}
 
+	count, cErr := a.Srv().Store().Reaction().GetUniqueCountForPost(reaction.PostId)
+	if cErr != nil {
+		return nil, model.NewAppError("SaveReactionForPost", "app.reaction.save.save.app_error", nil, "", http.StatusInternalServerError).Wrap(cErr)
+	}
+
+	if count >= *a.Config().ServiceSettings.UniqueEmojiReactionLimitPerPost {
+		return nil, model.NewAppError("SaveReactionForPost", "app.reaction.save.save.too_many_reactions", nil, "", http.StatusBadRequest)
+	}
```

Comment:
Yes I suppose so. We were investigating rate limiting, but it's not under the scope of my team and would likely be something we would want to enforce across multiple endpoints. Not in the scope of this PR I'm afraid.

Reply:
Would it make sense to pull that part out of this PR and focus on the primary issue of reacting with an emoji that doesn't exist? I wonder if we'll want to change our mind on the two separate features here and having a single commit might complicate that.

Reply:
We wanted to go the extra mile to fix this correctly, including making sure we can't overload the front-ends with multiple valid emoji reactions, since that is also a concern. I think we're fine to leave this as one piece.

Reply:
Deferring to the linked conversation on this topic -- thanks for the chance to provide feedback!